teps/0050-ignore-task-failures.md

```diff
@@ -187,6 +194,220 @@ control over the task definitions but may desire to ignore a failure and continu
+
+
+To distinguish pipeline run messages with and without ignored task failures, we explicitly add the ignored task failure count to ```pipelineRun.status.conditions.message``` in the following way if ignored task failure > 0: 
+
+```
+"Tasks Completed: A (Failed: B (C is ignored), Cancelled D), Skipped: E"
+```
+
+Example Input:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata: 
+  name: demo-pipeline-run
+spec:
+  pipelineSpec:
+    tasks:
+    - name: task1
+      onError: continue
+      taskSpec:
+        steps:
+          - image: alpine
+            name: exit-with-1
+            script: |
+              exit 1
+    - name: task2
+      taskSpec:
+        steps:
+          - image: alpine
+            name: exit-with-0
+            script: |
+              exit 0
+```
+
+Example Output:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+...
+status:
+  completionTime: "2022-08-15T17:26:15Z"
+  conditions:
+    - lastTransitionTime: "2022-08-15T17:26:15Z"
+      message: "Tasks Completed: 2 (Failed: 1 (1 is ignored), Cancelled 0), Skipped: 0"  
+      reason: Succeeded
+      status: "True"  # The failed task is considered "successful" when determining the state of pipelineRun
+      type: Succeeded
+  pipelineSpec:
+    ...
+  taskRuns:
+    demo-pipeline-run-task1:
+      pipelineTaskName: task1
+      status:
+        completionTime: "2022-08-15T17:26:13Z"
+        conditions:
+          - lastTransitionTime: "2022-08-15T17:26:13Z"
+            message: ...
+            reason: Failed
+            status: "False" # The task is failed even OnError is set to continue
+            type: Succeeded
+          ...
+    demo-pipeline-run-task2:
+      pipelineTaskName: task2
+      status:
+        completionTime: "2022-08-15T17:26:15Z"
+        conditions:
+          - lastTransitionTime: "2022-08-15T17:26:15Z"
+            message: All Steps have completed executing
+            reason: Succeeded
+            status: "True"
+            type: Succeeded
+          ...
+
+```
+
+### Ignored Failed Tasks with Retry
+Setting ```Retry``` and ```OnError``` to ```continue``` at the same time is not allowed, as there is no point to retry a task that allows to fail. Pipeline validation will be added accordingly.
+
+### Tasks with Missing Resource Dependency
+In the following example, the first task fails to produce a result (due to step error [details](https://github.com/tektoncd/pipeline/issues/3749)) that is going to be consumed by the second task. The second task will be skipped with reason ```Results were missing``` irrelevant of its OnError type.
```

Comment:
you can add a note that this behavior of skipping when there are missing results has a precedent:
https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guarding-a-task-only

and that this limitation can be addressed when we support default results as proposed in TEP-0050:
https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md

Reply:
References are added in this section